src/VisualStudio/Core/Def/ErrorReporting/VisualStudioInfoBar.cs

```diff
@@ -4,119 +4,175 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.VisualStudio;
 using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.ErrorReporting;
 
-internal sealed class VisualStudioInfoBar
+internal sealed class VisualStudioInfoBar(
+    IThreadingContext threadingContext,
+    IVsService<IVsInfoBarUIFactory> vsInfoBarUIFactory,
+    IVsService<IVsShell> vsShell,
+    IAsynchronousOperationListenerProvider listenerProvider,
+    IVsWindowFrame? windowFrame)
```

Comment:
part of https://github.com/dotnet/roslyn/pull/72765.  Will be gone when that PR goes in.